src/Core/Form/IdentifiableObject/CommandBuilder/Product/ProductCommandsBuilder.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\Product;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+
+/**
+ * This class builds a collection of product commands based on the form data and a list of ProductCommandBuilderInterface
+ */
+class ProductCommandsBuilder
+{
+    /**
+     * @var ProductCommandBuilderInterface[]
+     */
+    private $commandBuilders;
+
+    /**
+     * @param ProductCommandBuilderInterface[] $commandBuilders
+     */
+    public function __construct(array $commandBuilders)
+    {
+        $this->commandBuilders = $commandBuilders;
+    }
+
+    /**
+     * @param ProductId $productId
+     * @param array $formData
+     *
+     * @return ProductCommandCollection
+     */
+    public function buildCommands(ProductId $productId, array $formData): ProductCommandCollection
+    {
+        $commands = new ProductCommandCollection();
```

Comment:
This sounds a lot like an EditProduct saga.

https://medium.com/swlh/microservices-architecture-what-is-saga-pattern-and-how-important-is-it-55f56cfedd6b

Reply:
Mmm not exactly I think, because each command is independent The result of one has no impact on another and we don't have fallback implemented This is much simpler for now, it's just a sequence of commands, even their order is irrelevant

But is it a problem anyway? Do you think we should have an orchestrator to handle all this?